librarian_packages/hera_librarian/tests/test_utils.py

```diff
@@ -0,0 +1,176 @@
+
+    md5sum_padded = md5sum + "   "
+    assert utils.normalize_and_validate_md5(md5sum_padded) == md5sum
+
+    md5sum_upper = md5sum.upper() + "   "
+    assert utils.normalize_and_validate_md5(md5sum_upper) == md5sum
+
+    # make sure error is raised when length is incorrect
+    with pytest.raises(ValueError):
+        utils.normalize_and_validate_md5(md5sum[:-1])
+
+    return
+
+
+@ALL_FILES
+def test_md5_of_file(datafiles):
+    """Test generating md5sum of file"""
+    filepaths = list(map(str, datafiles.listdir()))
+    # make sure our files are ordered correctly
+    if filepaths[0].endswith("uvA"):
+        filepaths = filepaths[::-1]
+    assert utils._md5_of_file(filepaths[0]) == md5sums[0]
+
+    return
+
+
+@ALL_FILES
+def test_get_md5_from_path(datafiles):
+    """Test getting the md5sum for both a flat file and directory"""
+    filepaths = list(map(str, datafiles.listdir()))
+    # make sure our files are ordered correctly
+    if filepaths[0].endswith("uvA"):
+        filepaths = filepaths[::-1]
+    # test normal execution
+    for md5sum, path in zip(md5sums, filepaths):
+        assert utils.get_md5_from_path(path) == md5sum
+
+    # test adding funny bits to the ends of the directory names
+    datafile_miriad = filepaths[1] + "//."
+    assert utils.get_md5_from_path(datafile_miriad) == md5sums[1]
+
+    return
+
+
+@ALL_FILES
+def test_get_size_from_path(datafiles):
+    """Test computing filesize from path"""
+    filepaths = list(map(str, datafiles.listdir()))
+    # make sure our files are ordered correctly
+    if filepaths[0].endswith("uvA"):
+        filepaths = filepaths[::-1]
+    for pathsize, path in zip(pathsizes, filepaths):
+        assert utils.get_size_from_path(path) == pathsize
+
+    return
+
+
+@ALL_FILES
+def test_gather_info_for_path(datafiles):
+    """Test getting all info for a given path"""
+    filepaths = list(map(str, datafiles.listdir()))
+    # make sure our files are ordered correctly
+    if filepaths[0].endswith("uvA"):
+        filepaths = filepaths[::-1]
+    for filetype, md5, size, obsid, path in zip(
+        filetypes, md5sums, pathsizes, obsids, filepaths
+    ):
+        info = utils.gather_info_for_path(path)
+        assert info["type"] == filetype
+        assert info["md5"] == md5
+        assert info["size"] == size
+        assert info["obsid"] == obsid
+
+    return
+
+
+@ALL_FILES
+def test_print_info_for_path(datafiles, capsys):
+    """Test printing file info to stdout"""
+    filepaths = list(map(str, datafiles.listdir()))
+    # make sure our files are ordered correctly
+    if filepaths[0].endswith("uvA"):
+        filepaths = filepaths[::-1]
+    for filetype, md5, size, obsid, path in zip(
+        filetypes, md5sums, pathsizes, obsids, filepaths
+    ):
+        utils.print_info_for_path(path)
+        out, err = capsys.readouterr()
+        correct_string = '{{"obsid": {0:d}, "size": {1:d}, "type": "{2:}", "md5": "{3:}"}}'.format(
```

Comment:
Maybe parse the JSON here? The order of the keys in the dictionary might change, or something ...

Reply:
Didn't think of that! I've done this now.